libpromises/cf3parse.y

```diff
@@ -755,10 +755,23 @@ litem:                 IDENTIFIER
                                        (P.currentclasses ?
                                             P.currentclasses : "any"),
                                        P.currentstring);
-                           RlistAppendScalar((Rlist **) &P.currentRlist,
-                                             (void *) P.currentstring);
-                           free(P.currentstring);
-                           P.currentstring = NULL;
+
+                           assert(!NULL_OR_EMPTY(P.lval));
+
+                           if (!StringEqual(P.lval + 1, "list") || P.lval[0] == 's')  // slists and everything else
+                           {
+                               RlistAppendScalar((Rlist **) &P.currentRlist,
+                                                 (void *) P.currentstring);
+                               FREE_AND_NULL(P.currentstring);
+                           }
+                           else if (P.lval[0] == 'i')  // ilist
+                           {
+                               ParserHandleQuotedIlistItems();
+                           }
+                           else // rlist
+                           {
```

Comment:
It's better to add asserts instead of these `ilist/rlist` comments. They bring more value than just the comments.